iothub/device/src/NumericHelpers.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    internal class NumericHelpers
+    {
+        internal static bool TryCastNumericTo<T>(object input, out T result)
```

Comment:
NewtonSoft will deserialize a float into a double (long to int, etc), at which point the `TryGetValue` API with the higher range value type will fail. This helper is added to handle those numeric conversions.